packages/lodestar/src/db/api/beacon/repositories/blockArchive.ts

```diff
@@ -22,6 +23,42 @@ export class BlockArchiveRepository extends Repository<Slot, SignedBeaconBlock>
     super(config, db, Bucket.blockArchive, config.types.SignedBeaconBlock);
```

Comment:
Can you extend the whole interface for completeness?
Will be helpful for eg: a db sniffer utility. And there's nothing warning people from using methods that don't respect the secondary index.

Reply:
What methods does not respect secondary index?

Cannot really extend delete methods since they only receive slot but it doesn't really matter if it points to non existent slot, it will return null anyways.

Reply:
`put`, `delete`, `remove`, `batchPut`, `batchDelete`, `batchRemove`

Reply:
I should probably override put and batchPut instead of add and batchAdd. remove/delete/batchDelete/batchRemove are useless to overide

Reply:
remove / batchRemove is useful (and easy to implement) for db diagnostic tooling. We should have it for completeness, and because there is no warning to users that it won't respect the secondary index.

Hypothetical:
- synced a lot of blocks
- there's a bug in our state transition
- we want to "remove" a finalized block from the db

Other hypothetifcal:
- synced a lot of blocks
- we figure out a better on-disk format / db for storing _very old_ blocks but not recently old blocks
- we want to delete the oldest blocks and move them to another db